File: src/Qwack.Transport/TransportObjects/MarketData/Models/TO_FxMatrix.cs
using System;
using System.Collections.Generic;
using ProtoBuf;

namespace Qwack.Transport.TransportObjects.MarketData.Models
{
    [ProtoContract]
    public class TO_FxMatrix
    {
        [ProtoMember(2)]
        public string BaseCurrency { get; set; }
        [ProtoMember(3)]
        public DateTime BuildDate { get; set; }
        [ProtoMember(4)]
        public List<TO_FxPair> FxPairDefinitions { get; set; }
        [ProtoMember(5)]
        public Dictionary<string, string> DiscountCurveMap { get; set; }
        [ProtoMember(6)]
        public Dictionary<string, double> SpotRates { get; set; }
    }
}

File: src/Qwack.Transport/GeneratedTypeScript/TO_FxMatrix.ts

import { TO_FxPair } from './TO_FxPair';

export interface TO_FxMatrix  {
	 baseCurrency: string;
	 buildDate: Date;
	 fxPairDefinitions: TO_FxPair[];
	 discountCurveMap: { [key: string]: string; };
	 spotRates: { [key: string]: number; };
}


